pkg/handlers/upgrade_service.go

```diff
@@ -75,6 +75,16 @@ func (h *Handler) StartUpgradeService(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if foundApp.IsAirgap && foundApp.ChannelID != request.ChannelID {
+		// The update is deployable, so we should update the apps stored channelID similar to install
+		// time when channelID is set via the requested-channel-id in the configmap.
+		if err := store.GetStore().SetAppChannelID(foundApp.ID, request.ChannelID); err != nil {
```

Comment:
setting the app channel id should happen [here](https://github.com/replicatedhq/kots/blob/main/pkg/operator/operator.go#L1024) instead, as that's when the update gets deployed, not here. that's because the user might decide to abort the upgrade and not actually deploy this update.